corehq/apps/hqcase/management/commands/backfill_couch_forms_and_cases.py

```diff
@@ -0,0 +1,201 @@
+        for domain, form_ids in all_form_ids_by_domain.items():
+            missing_form_ids_by_domain[domain] = use_json_cache_file(
+                filename='missing_form_ids_{}_to_{}__{}.json'.format(start, end, domain),
+                fn=lambda: get_form_ids_missing_from_elasticsearch(form_ids)
+            )
+        print("[3] Get all case ids by domain for date range")
+        all_case_ids_by_domain = use_json_cache_file(
+            filename='all_case_ids_last_modified_{}_to_{}_by_domain.json'.format(start, end),
+            fn=lambda: get_all_case_ids_by_domain(start, end)
+        )
+        print("[4] Get case ids by domain missing from ES")
+        missing_case_ids_by_domain = {}
+        for domain, case_ids in all_case_ids_by_domain.items():
+            if case_ids:
+                missing_case_ids_by_domain[domain] = use_json_cache_file(
+                    filename='missing_case_ids_{}_to_{}__{}.json'.format(start, end, domain),
+                    fn=lambda: get_case_ids_missing_from_elasticsearch(case_ids)
+                )
+
+        print("[5] Get all the _revs for these docs")
+        case_metadata = use_json_cache_file(
+            filename='missing_case_metadata_{}_to_{}.json'.format(start, end),
+            fn=lambda: prepare_metadata(missing_case_ids_by_domain)
+        )
+        form_metadata = use_json_cache_file(
+            filename='missing_form_metadata_{}_to_{}.json'.format(start, end),
+            fn=lambda: prepare_metadata(missing_form_ids_by_domain)
+        )
+
+        print("[6] Publish changes for docs missing from ES!")
+        interleaved_changes = (change for pair in zip_longest(
+            iter_case_changes(case_metadata),
+            iter_form_changes(form_metadata),
+        ) for change in pair if change is not None)
+        for changes in chunked(interleaved_changes, 100):
+            for change in changes:
+                publish_change(change)
+            time.sleep(1)
+
+
+def generate_all_form_ids_by_domain(start, end):
+    form_ids_by_domain = {
+        domain: get_form_ids_by_type(domain, 'XFormInstance', start, end)
+        for domain in iter_domains()
+    }
+    return {
+        domain: form_ids
+        for domain, form_ids in form_ids_by_domain.items()
+        if form_ids
+    }
+
+
+def get_form_ids_missing_from_elasticsearch(all_form_ids):
+    missing_from_elasticsearch = set()
+    for form_ids in chunked(all_form_ids, 500):
+        form_ids = set(form_ids)
+        not_missing = set(FormES().doc_id(form_ids).get_ids())
+        missing_from_elasticsearch.update(form_ids - not_missing)
+        assert not_missing - form_ids == set()
+    return list(missing_from_elasticsearch)
+
+
+def get_case_ids_missing_from_elasticsearch(all_case_ids):
+    missing_from_elasticsearch = set()
+    for case_ids in chunked(all_case_ids, 500):
+        case_ids = set(case_ids)
+        not_missing = set(CaseES().doc_id(case_ids).get_ids())
```

Comment:
do you know how this compares to using `mget`?

Reply:
I'm not sure, how it compares performance-wise, but there didn't seem to be any performance issues doing it this way; of all the things in this command, this one was rather quick